packages/mask/background/services/backup/internal_wallet_restore.ts

```diff
@@ -20,7 +20,17 @@ export async function internal_wallet_restore(backup: NormalizedBackup.WalletBac
     for (const wallet of backup) {
         try {
             const wallets = await getWallets()
-            const name = wallet.name || generateNewWalletName(wallets)
+            let nameExistsIndex = 0
```

Comment:
An easier way: parse the index of every wallet name. Compute the max index among them and the next qualified index is the max index + 1.